adhocracy4/follows/static/follows/__tests__/FollowButton.jest.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { render, fireEvent, screen } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { FollowButton } from '../FollowButton'
+import jQuery from 'jquery'
+import api from '../../../../static/api'
+window.$ = window.jQuery = jQuery
+
+jest.mock('../../../../static/config')
+jest.mock('../../../../static/api')
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+test('Test render FollowButton not following', async () => {
+  api.follow.setFollowing({ enabled: false })
+  render(<FollowButton authenticatedAs project="test" />)
+  const followButton = await screen.findByText('Follow')
+  expect(followButton).toBeTruthy()
+  const followingButton = screen.queryByText('Following')
+  expect(followingButton).toBeNull()
+})
+
+test('Test render FollowButton following', async () => {
+  api.follow.setFollowing({ enabled: true })
+  render(<FollowButton authenticatedAs project="test" />)
+  const followingButton = await screen.findByText('Following')
+  expect(followingButton).toBeTruthy()
+  const followButton = screen.queryByText('Follow')
+  expect(followButton).toBeNull()
+})
+
+test('Test render FollowButton click follow', async () => {
+  api.follow.setFollowing({ enabled: false })
+  render(<FollowButton authenticatedAs project="test" />)
+  let followButton = await screen.findByText('Follow')
+  expect(followButton).toBeTruthy()
+  let followingButton = screen.queryByText('Following')
+  expect(followingButton).toBeNull()
+  fireEvent.click(followButton)
+  followingButton = await screen.findByText('Following')
+  expect(followingButton).toBeTruthy()
+  followButton = screen.queryByText('Follow')
+  expect(followButton).toBeNull()
+  expect(api.follow.change).toHaveBeenCalledTimes(1)
+})
+
+test('Test FollowButton redirect', async () => {
+  delete window.location
+  window.location = {}
```

Comment:
what are these 2 lines doing? maybe a comment would be good unless its very obvious and i'm just missing it?!

Reply:
will add comment